jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java

```diff
@@ -31,167 +30,115 @@
 import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
 import io.fabric8.kubernetes.api.model.apps.Deployment;
 import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
 import io.fabric8.openshift.api.model.DeploymentConfigBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
 
 /**
  * Test label generation.
  *
  * @author Tue Dissing
  */
-public class ProjectLabelEnricherTest {
-  private JKubeEnricherContext context;
+class ProjectLabelEnricherTest {
 
   private Properties properties;
   private ProjectLabelEnricher projectLabelEnricher;
 
-  @Before
-  public void setupExpectations() {
-    context = mock(JKubeEnricherContext.class, RETURNS_DEEP_STUBS);
+  @BeforeEach
+  void setupExpectations() {
+    JKubeEnricherContext context = mock(JKubeEnricherContext.class, RETURNS_DEEP_STUBS);
     projectLabelEnricher = new ProjectLabelEnricher(context);
     properties = new Properties();
     when(context.getProperties()).thenReturn(properties);
     when(context.getGav()).thenReturn(new GroupArtifactVersion("groupId", "artifactId", "version"));
   }
 
   @Test
-  public void testCustomAppName() {
-    // Setup
+  void create_customAppName() {
+    // Given
     properties.setProperty("jkube.enricher.jkube-project-label.app", "my-custom-app-name");
-
-    KubernetesListBuilder builder = createListWithDeploymentConfig();
-    projectLabelEnricher.enrich(PlatformMode.kubernetes, builder);
-    KubernetesList list = builder.build();
-
-    Map<String, String> labels = list.getItems().get(0).getMetadata().getLabels();
-    assertThat(labels).isNotNull()
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","my-custom-app-name")
-            .containsEntry("version","version");
-
-
-    builder = new KubernetesListBuilder().withItems(new DeploymentBuilder().build());
+    KubernetesListBuilder builder = new KubernetesListBuilder().withItems(new DeploymentBuilder().build());
+    // When
     projectLabelEnricher.create(PlatformMode.kubernetes, builder);
-
+    // Then
     Deployment deployment = (Deployment) builder.buildFirstItem();
-    Map<String, String> selectors = deployment.getSpec().getSelector().getMatchLabels();
-    assertThat(selectors)
-            .doesNotContainKey("version")
-            .doesNotContainKey("project")
-            .containsEntry("group","groupId")
-            .containsEntry("app","my-custom-app-name");
+    assertThat(deployment)
+        .extracting("spec.selector.matchLabels")
+        .asInstanceOf(InstanceOfAssertFactories.MAP)
+        .containsEntry("group", "groupId")
+        .containsEntry("app", "my-custom-app-name")
+        .doesNotContainKey("version")
+        .doesNotContainKey("project");
   }
 
   @Test
-  public void testEmptyCustomAppName() {
-    // Setup
+  void create_emptyCustomAppName() {
```

Comment:
You need to create an additional test for DeploymentConfig, the test you refactored was testing both Deployment and DeploymentConfig